src/components/forms/live-input-hoc.jsx

```diff
@@ -0,0 +1,60 @@
+import bindAll from 'lodash.bindall';
+import PropTypes from 'prop-types';
+import React from 'react';
+
+/**
+ * Higher Order Component to manage inputs that submit on change and <enter>
+ * @param {React.Component} Input text input that consumes onChange, onBlur, onKeyPress
+ * @returns {React.Component} Live input that calls onSubmit on change and <enter>
+ */
+export default function (Input) {
+    class LiveInput extends React.Component {
+        constructor (props) {
+            super(props);
+            bindAll(this, [
+                'handleChange',
+                'handleKeyPress',
+                'handleFlush'
+            ]);
+            this.state = {
+                value: null
+            };
+        }
+        handleKeyPress (e) {
```

Comment:
Nit: The function names handleChange, handleKeyPress, and handleFlush all kind of sound like the same thing, so it's hard to guess which one will be called by what and what order they'll happen in. Maybe the default wording on an input is just confusing. Do onKeyPress and onChange always both happen? Maybe we can come up with better names?